### DISCUSSION TITLE
Return TXN ID after deploying contract

### DISCUSSION THREAD

**Discussion by CoffeeBoi-1** [reach-sh/reach-lang#984] [Ideas]
My idea is return TXN ID (or more information) after contact deployed and approved
`let txnId = await p.contract(backend)`

**Comment by jeapostrophe** [comment]
`acc.contract` returns a contract handle --- https://docs.reach.sh/frontend/#js_contract --- and it does not deploy the contract.

What is the semantics of a "TXN ID"?

**Reply by CoffeeBoi-1** [reply]
I mean transaction ID like that :
`https://testnet.algoexplorer.io/tx/J4FEE2L26PHWL7JIXJCQMDXKA6OKPPWBV4SCT4BLYX75X4WC2NSQ`

I need some function, that returns ID of that transaction
![image](https://user-images.githubusercontent.com/51762237/168852118-e8b733d9-19c4-4b52-9542-af7de070887d.png)

**Reply by jeapostrophe** [reply]
What do you use this value for?

**Reply by CoffeeBoi-1** [reply]
I need it for my approvable random number algorithm

**Reply by jeapostrophe** [reply]
I don't understand. Your user already has access to it, because they see it in their wallet display. But you want programmatic access so you can look at the bytes and use them to generate a "random" number?

**Reply by CoffeeBoi-1** [reply]
What the bytes? Where can I get it?

**Reply by jeapostrophe** [reply]
I'm asking you what you're using them for. Are you saying you want to use the txn id bytes as input to a random number generation algorithm?

**Reply by CoffeeBoi-1** [reply]
Yes, that's it

**Reply by jeapostrophe** [reply]
That is very insecure. Are you sure you want to do it? Why not just generate cryptographically secure random bytes locally?

**Reply by CoffeeBoi-1** [reply]
Generate it by reach random function?
How does this random function works?
But partipiant can modify library and send not random number

**Reply by jeapostrophe** [reply]
Yes, but the transaction id isn't random either. And someone could modify _your_ program and just not send it (transaction ids are not visible on-chain.) The only safe way to get randomness on-chain is to have adversaries each choose different values and combine them.

**Reply by CoffeeBoi-1** [reply]
I need transaction ID, because anyone can approve, that random number based on transaction ID is not fake
And we don't know when program gonna be approved on-chain, so txn ID - is random?

EDIT
How txn ID generates?

**Reply by jeapostrophe** [reply]
It actually doesn't work like that. The txn id is not random and can be controlled and influenced by users and nodes. Furthermore, if the on-chain program is going to use it, then that means a participant needs to publish it and there's no way for the on-chain program to check that it really is a txn id.

**Reply by CoffeeBoi-1** [reply]
Program does not need to check
After everything, users gonna check txn id
